cli/azd/CHANGELOG.md

```diff
@@ -5,6 +5,7 @@
 ### Bugs Fixed
 
 - [[2402]](https://github.com/Azure/azure-dev/pull/2279) Support for workload profiles in Azure Container Apps
+- [[2428, 2040]](https://github.com/Azure/azure-dev/pull/2468) Include current git branch in GitHub federated credentials
```

Comment:
Can we update this to re-use the SP that was already used?

For example.

Let's say I run this on main, then i want to run it on 'dev' branch.  As it is now, I think multiple SPs, Apps, etc.  will be created.  Can we reuse the existing ones?

Reply:
You can use `--principal-name foo` to re-use a `foo` SP (with some bottom notes).

* If you re-use a SP using client-secret, the secret is rotated and any current repository using that SP is broken, as it has previous credentials

Reply:
@jongio - I agree with @vhvb1989 here.  Devs should tell azd with principal to use which would go ahead and then just add the new federated auth credential into the existing identity.  We do store the `AZURE_CLIENT_ID` of the identity in azd env but this may not always be the desired flow in the event they wanted to use different identities for different branches.

Happy to take this as a follow-up change if we change our mind.